lib/gringotts/gateways/pin_pay.ex

```diff
@@ -0,0 +1,423 @@
+
+
+  > PINPAY supports more optional keys and you can raise an [issue][issues] if
+    this is important to you.
+
+  [issues]: https://github.com/aviabird/gringotts/issues/new
+
+
+  ### Schema
+
+  * `address` is a `map` from `atoms` to `String.t`, and can include any
+    of the keys from:
+    `[:street1, :street2, :city, :region, :postal_code, :country, :phone, ]`
+
+
+  ## Registering your PINPAY account at `Gringotts`
+
+  | Config parameter | PINPAY secret       |
+  | -------          | ----              |
+  | `:username`      | **API_SECRET_KEY**  |
+  | `:password`      | Empty string        |
+
+  > Your Application config **must include the `:username`, `:password`
+  > fields** and would look something like this:
+
+      config :gringotts, Gringotts.Gateways.Pinpay,
+          username: "your_secret_key",
+          password: "",
+
+  * PINPAY **does not** process money in cents.
+  * Although PINPAY supports payments various cards. This module only
+  accepts payments via `VISA`, `MASTER`, and `AMERICAN EXPRESS`.
+
+  ## Supported countries
+  PINPAY supports the countries listed [here][all-country-list]
+  $ AUD, $ USD, $ NZD, $ SGD, € EUR, £ GBP, $ CAD, ¥ JPY
+
+  ## Supported currencies
+  PINPAY supports the currencies listed [here][all-currency-list]
+  :AUD, :USD, :NZD, :SGD, :EUR, :GBP, :CAD, :HKD, :JPY, :MYR, :THB, :PHP, :ZAR, :IDR, :TWD
+
+
+  ## Following the examples
+
+  1. First, set up a sample application and configure it to work with Monei.
+  - You could do that from scratch by following our [Getting Started][gs] guide.
+      - To save you time, we recommend [cloning our example
+      repo][example] that gives you a pre-configured sample app ready-to-go.
+          + You could use the same config or update it the with your "secrets"
+          as described [above](#module-registering-your-monei-account-at-PinPay).
+
+  2. Run an `iex` session with `iex -S mix` and add some variable bindings and
+  aliases to it (to save some time):
+  ```
+  iex> alias Gringotts.{Response, CreditCard, Gateways.Pinpay}
+  iex> card = %CreditCard{first_name: "Jo",
+                          last_name: "Doe",
+                          number: "4200000000000000",
+                          year: 2099, month: 12,
+                          verification_code: "123", brand: "VISA"}
+  ```
+
+  > Add any other frequently used bindings up here.
+
+  We'll be using these in the examples below.
+
+  [gs]: https://github.com/aviabird/gringotts/wiki/
+  [home]: https://pinpayments.com
+  [docs]: https://pinpayments.com/developers/api-reference
+  [example]: https://github.com/aviabird/gringotts_example
+  """
+
+  # The Base module has the (abstract) public API, and some utility
+  # implementations.  
+  use Gringotts.Gateways.Base
+
+  # The Adapter module provides the `validate_config/1`
+  # Add the keys that must be present in the Application config in the
+  # `required_config` list
+  use Gringotts.Adapter, required_config: []
```

Comment:
Please do as suggested [here](https://aviabird.slack.com/archives/C98HQMNPR/p1521564554000727)

Reply:
Why is `required_config` empty?